src/dex/angle-staked-stable/angle-staked-stable.ts

```diff
@@ -0,0 +1,314 @@
+
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly needWrapNative = true;
+
+  readonly isFeeOnTransferSupported = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(AngleStakedStableConfig);
+
+  logger: Logger;
+
+  protected eventPools: { [key: string]: AngleStakedStableEventPool } = {};
+  protected isPaused: { [key: string]: boolean } = {};
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    protected adapters = Adapters[network] || {},
+  ) {
+    super(dexHelper, dexKey);
+    const config = AngleStakedStableConfig[dexKey][network];
+    this.logger = dexHelper.getLogger(dexKey);
+    this.config = {
+      stakeToken: config.stakeToken.toLowerCase(),
+      agToken: config.agToken.toLowerCase(),
+    };
+  }
+
+  async initializePricing(blockNumber: number) {
+    this.eventPools[this.config.stakeToken] = new AngleStakedStableEventPool(
+      `${this.dexKey}_${this.config.stakeToken!.toLowerCase()}`,
+      this.network,
+      this.dexHelper,
+      this.config.stakeToken,
+      this.config.agToken,
+      this.logger,
+    );
+    await this.eventPools[this.config.stakeToken].initialize(blockNumber);
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const knownInfo = this._knownAddress(srcToken, destToken);
+    if (!knownInfo.known) return [];
+    return [`${this.dexKey}_${knownInfo.stakeToken!.toLowerCase()}`];
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<AngleStakedStableData>> {
+    const srcTokenAddress = srcToken.address.toLowerCase();
+    const destTokenAddress = destToken.address.toLowerCase();
+    const knownInfo = this._knownAddress(srcToken, destToken);
+    if (!knownInfo.known) return null;
+
+    const agToken = knownInfo.agToken!;
+    const stakeToken = knownInfo.stakeToken!;
+    const eventPool = this.eventPools[stakeToken];
+    const exchange = `${this.dexKey}`;
+    const state = eventPool?.getState(blockNumber);
+    if (this.eventPools === null || state === undefined || state === null)
+      return null;
+    if (srcTokenAddress === agToken && side === SwapSide.SELL)
+      return [
+        {
+          prices: amounts.map(amount =>
+            eventPool.getRateDeposit(amount, state),
+          ),
+          unit: eventPool.getRateDeposit(1n * BigInt(10 ** 18), state),
+          gasCost: AngleStakedGasCost,
+          exchange: exchange,
+          data: { exchange: `${stakeToken}` },
+          poolAddresses: [`${stakeToken}`],
```

Comment:
typo: probably should be `${stakeToken}_${agToken}`?

Reply:
it's not used here, so I'll just remove it

Reply:
Fixed to have the right address for all cases